py/server/deephaven/agg.py

```diff
@@ -201,31 +202,39 @@ def max_(cols: Union[str, List[str]] = None) -> Aggregation:
     return Aggregation(j_agg_spec=_JAggSpec.max(), cols=cols)
 
 
-def median(cols: Union[str, List[str]] = None) -> Aggregation:
-    """Create a Median aggregation.
+def median(cols: Union[str, List[str]] = None, average_evenly_divided: bool = True) -> Aggregation:
+    """Create a Median aggregation which computes the median value within an aggregation group for each of the
+    given column.
 
     Args:
         cols (Union[str, List[str]]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
             default is None, only valid when used in Table agg_all_by operation
+        average_evenly_divided (bool): when the group size is an even number, whether to average the two middle values
+            for the output value, default is True. when set to False, use the smaller value. This flag is only valid for
+            numeric types.
 
     Returns:
         an aggregation
     """
-    return Aggregation(j_agg_spec=_JAggSpec.median(), cols=cols)
+    return Aggregation(j_agg_spec=_JAggSpec.median(average_evenly_divided), cols=cols)
 
 
-def pct(percentile: float, cols: Union[str, List[str]] = None) -> Aggregation:
-    """Create a Percentile aggregation.
+def pct(percentile: float, cols: Union[str, List[str]] = None, average_evenly_divided: bool = False) -> Aggregation:
+    """Create a Percentile aggregation which computes the percentile value within an aggregation group for each of
+    the given columns.
 
     Args:
         percentile (float): the percentile used for calculation
         cols (Union[str, List[str]]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
             default is None, only valid when used in Table agg_all_by operation
+        average_evenly_divided (bool): when the percentile is 0.5 and the group size is an even number, whether to
```

Comment:
This is not specific to percentile == 0.5

Reply:
Hmm, then the Java doc should be revised? 

```
   /**
     * The percentile to calculate. Must be &gt;= 0.0 and &lt;= 1.0.
     *
     * @return The percentile to calculate
     */
    @Parameter
    public abstract double percentile();

    /**
     * Whether to average the highest low-bucket value and lowest high-bucket value, when the low-bucket and high-bucket
     * are of equal size. Only applies to numeric types.
     *
     * @return Whether to average the two result candidates for evenly-divided input sets of numeric types
     */
    @Parameter
    public abstract boolean averageEvenlyDivided();

```

Reply:
Where does the JavaDoc say the language I objected to?

Reply:
`when the low-bucket and high-bucket
     * are of equal size.`

Reply:
This is about the implementation details. It's a more reasonable consideration for 50th percentile, but in cases with relatively few inputs, we could still be evenly divided at other percentiles.
The formula for how big the low bucket will be is:
```
            final long targetLo = (int)((totalSize - 1) * percentile) + 1;
```
So if you have 2 values and 70th percentile, we'll be evenly divided. Same for 4 values and 60th percentile.